plugins/http_plugin/http_plugin.cpp

```diff
@@ -271,6 +272,13 @@ namespace eosio {
             boost::filesystem::path sock_path = options.at("unix-socket-path").as<string>();
             if (sock_path.is_relative())
                sock_path = app().data_dir() / sock_path;
+            // The maximum length of the socket path is defined by sockaddr_un::sun_path. On Linux,
+            // according to unix(7), it is 108 bytes. On FreeBSD, according to unix(4), it is 104 bytes.
+            const auto unix_socket_limit = sizeof(sockaddr_un::sun_path);
+            if (sock_path.string().size() >= unix_socket_limit) {
+               // try a relative path which may be shorter.
+               sock_path = boost::filesystem::relative(sock_path);
+            }
```

Comment:
The motivation of the PR is to avoid testing failing when the test log directories are too long to begin with. Just let user to re-try with a relative path doesn't help the problem.

Reply:
I like Matt's idea. Then it would always work the same way and could never fail do to length issue.